metricflow/sql/render/redshift.py

```diff
@@ -59,6 +65,31 @@ def visit_percentile_expr(self, node: SqlPercentileExpression) -> SqlExpressionR
             bind_parameters=params,
         )
 
+    @override
+    def render_date_part(self, date_part: DatePart) -> str:
+        return date_part.value
+
+    @override
+    def visit_extract_expr(self, node: SqlExtractExpression) -> SqlExpressionRenderResult:
+        """Renders extract expressions with required output conversions for Redshift.
+
+        Redshift does not have native support for the ISO standard day of week output of 1 (Monday) - 7 (Sunday).
+        Instead, Redshift returns 0 (Sunday) - 6 (Monday). Therefore, we need custom rendering logic to normalize
+        the return values to the ISO standard.
+        """
+        extract_rendering_result = super().visit_extract_expr(node)
+
+        if node.date_part is not DatePart.DOW:
+            return extract_rendering_result
+
+        extract_stmt = extract_rendering_result.sql
+        case_expr = f"CASE WHEN {extract_stmt} = 0 THEN {extract_stmt} + 7 ELSE {extract_stmt} END"
```

Comment:
I used `IF` because it's shorter, but it's not ANSI standard so not all engines support it. That's probably a mistake, maybe I'll update it before merging.

As for the SqlExpressionNodes - yes, long term I think that's what we should do. We can have a SqlCaseExpressionNode that does nicely formatted case statements, that seems broadly useful. I didn't bother here because I couldn't figure out how to cleanly transform the extract node without doing it outside of the rendering layer. There are a bunch of options to think through.

Reply:
Ok, after getting annoyed by the visitor pattern yet again (I think I really went over the edge with #819 ) I've come to the conclusion that using a CASE expression node has to happen inside the  renderer.

I'm going to just merge this as-is and then I'll follow up with a CASE statement node and see if I can hijack the rendering accordingly. If I can't make it nicely formatted I'll just stick with what's here until we push an update to reformat our SQL to dbt-like standards - given the bad formatting we have already the shorter IF seems fine for BigQuery even if it's not consistent with the CASE/WHEN we need to use for other engines.